control-plane/pkg/kafka/consumer_group_lag.go

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright 2021 The Knative Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kafka
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/Shopify/sarama"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+const (
+	invalidOffset = -1
+)
+
+// ConsumerGroupLagProvider provides consumer group lags.
+type ConsumerGroupLagProvider interface {
+	// GetLag returns consumer group lag for a given topic and a given consumer group.
+	GetLag(topic, consumerGroup string) (ConsumerGroupLag, error)
+
+	// Close closes the consumer group lag provider.
+	Close() error
+}
+
+// ConsumerGroupLag contains partition lag of a topic.
+type ConsumerGroupLag struct {
+	Topic         string
+	ConsumerGroup string
+	ByPartition   []PartitionLag
+}
+
+// PartitionLag contains consumer lag information of a partition.
+type PartitionLag struct {
```

Comment:
Maybe rename `PartitionOffsetStatus`?

Reply:
It's pretty common in the Kafka world `Lag`, in fact, this command:
```./bin/kafka-consumer-groups.sh --bootstrap-server localhost:9092 --group mygroup  --describe```
produces an output like this:
```
TOPIC                          PARTITION  CURRENT-OFFSET  LOG-END-OFFSET  LAG        CONSUMER-ID                                       HOST
```

Reply:
Lag in the output is exactly `PartitionLag` in our code.

Reply:
Ok I wasn't aware of that, i'm sorry, then lag it is

Reply:
No problem!